File: RecommendationEngine/Models/Application/ConfiguredRecommendation.cs
using System;
using System.Collections.Generic;
using Models.Application.Asset;

namespace Models.Application
{
    public class ConfiguredRecommendation
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Type { get; set; }
        public string Description { get; set; }
        public string Granularity { get; set; }
        public string CreatedBy { get; set; }
        public string PreferredScenario { get; set; }
        public int RecurrenceDayOfWeek { get; set; }
        public DateTime RecurrenceDatetime { get; set; }
        public DateTime CreatedOn { get; set; }
        public List<ConfiguredRecommendationJob> LastJobs { get; set; }
        public List<int> AssetIdList { get; set; }
        public List<AssetLeaf> AssetList { get; set; }
        public List<ConfiguredRecommendationParameter> Parameters { get; set; }
    }
}

File: RecommendationEngine/RecommendationEngine/FrontEnd/src/api/models/ConfiguredRecommendation.ts
//You have to make sure that the method that fetches this object
//have exactly this attribute, or it will break and it will be hard to debug.
import { Asset } from './Asset';
import { Parameter } from './TemplateDetails';
export interface ConfiguredRecommendation {
    id: number,
    name: string,
    type: string,
    description: string,
    granularity: string,
    createdBy: string,
    preferredScenario: string,
    recurrenceDayOfWeek: number,
    recurrenceDatetime: string,
    createdOn: string,
    lastJobs: Job[],
    assetIdList?: number[],
    assetList?: Asset[],
    parameters?: Parameter[]
}
export interface Job {
    id: number,
    status: string
    timestamp: Date
}